src/algorithm/length.rs

```diff
@@ -0,0 +1,82 @@
+use num::Float;
+
+use types::{LineString, MultiLineString};
+use algorithm::distance::Distance;
+
+/// Calculation of the length
+
+pub trait Length<T, RHS = Self> {
+    /// Calculation the length of a Line
+    ///
+    /// ```
+    /// use geo::{Point, LineString, Coordinate};
+    /// use geo::algorithm::length::Length;
+    ///
+    /// let mut vec = Vec::new();
+    /// vec.push(Point::new(40.02f64, 116.34));
+    /// vec.push(Point::new(42.02f64, 116.34));
+    /// let linestring = LineString(vec);
+    ///
+    /// println!("Length {}", linestring.length());
+    /// ```
+    ///
+    fn length(&self) -> T;
+}
+
+impl<T> Length<T> for LineString<T>
+    where T: Float
+{
+    ///
+    /// Length of a LineString.
+    ///
+    fn length(&self) -> T {
+        let vect = &self.0;
+        if vect.is_empty() || vect.len() == 1 {
+            return T::zero();
+        } else {
+            let ipoints = vect.iter().zip(vect[1..].iter());
+            ipoints.fold(T::zero(), |total_length, (p1, p2)| total_length + p1.distance(&p2))
+        }
+    }
```

Comment:
This might be simpler if you used [`slice::windows`](https://doc.rust-lang.org/std/primitive.slice.html#method.windows) here.